tests/test_cli_compile.py

```diff
@@ -1476,6 +1477,10 @@ def test_multiple_input_files_without_output_file(runner):
     assert out.exit_code == 2
 
 
+@pytest.mark.xfail(
+    Version(pip.__version__) >= Version("24.3"),
+    reason="known issue in newer versions of pip (#2110)",
```

Comment:
The message should be clear about what versions are bad so that it's more obvious when it can be dropped. This is a part of the test run output to the terminal, saying that some arbitrary pip version that was newer at an arbitrary point in time in the past is unhelpful.